### PR TITLE
Refine create_runtime_config()

### PR DESCRIPTION
**Pull request by jiangliu**
Refine create_runtime_config() by:
1) add more comments
2) fix error in generating annotations
3) add support for missed annotations

### CODE REVIEW

**Review by sameo — APPROVED** [review]

**Review by sameo — APPROVED** [review]
LGTM, let's wait for @arronwy review before merging.

File: src/bundle.rs [hunk 1]
```diff
@@ -14,66 +14,134 @@ pub const BUNDLE_ROOTFS: &str = "rootfs";
...
+/// runtime configuration".
 pub fn create_runtime_config(
     image_config: &ImageConfiguration,
     bundle_path: &Path,
 ) -> Result<PathBuf> {
     let mut spec = Spec::default();
+    let mut annotations: HashMap<String, String> = HashMap::new();
+    let mut labels = HashMap::new();
 
+    // Update the default hostname of `youki`
```
Comment (arronwy) [inline src/bundle.rs:44]: a small typo of hostname fo `image-rs`

File: src/bundle.rs [hunk 2]
```diff
@@ -14,66 +14,134 @@ pub const BUNDLE_ROOTFS: &str = "rootfs";
...
             args.extend(entrypoint.clone());
         }
-
         if let Some(cmd) = config.cmd() {
             args.extend(cmd.clone());
         }
-
         if !args.is_empty() {
             process.set_args(Some(args));
         }
```
Comment (arronwy) [inline src/bundle.rs:79]: There have a bug in original code, could help fix it together with this PR?
we need update the `process` in `spec` which is missing in original code :-(
```
spec.set_process(Some(process));
```

**Review by arronwy — COMMENTED** [review]
Thanks @jiangliu , LGTM, only some minor changes maybe needed.

File: src/bundle.rs (hunk 2 above)
Comment (jiangliu) [reply src/bundle.rs:79]: Yes, I also found this issue, but missed it.

**Review by jodh-intel — APPROVED** [review]
Nice PR @jiangliu - thanks !

File: src/bundle.rs (hunk 1 above)
Comment (jiangliu) [reply src/bundle.rs]: Fixed

**Pull request merged**